Tests/AsyncQueueTests/AsyncQueueTests.swift

```diff
@@ -26,8 +26,202 @@ import XCTest
+        systemUnderTest = nil
+        // Signal the semaphore to unlock the remaining enqueued tasks.
+        await semaphore.signal()
+
+        await waitForExpectations(timeout: 1.0)
+    }
+
+    func test_async_doesNotRetainTaskAfterExecution() async {
+        final class Reference: Sendable {}
+        final class ReferenceHolder: @unchecked Sendable {
+            var reference: Reference? = Reference()
+        }
+        let referenceHolder = ReferenceHolder()
+        weak var weakReference = referenceHolder.reference
+        let asyncSemaphore = Semaphore()
+        let syncSemaphore = Semaphore()
+        systemUnderTest.async { [reference = referenceHolder.reference] in
+            // Now that we've started the task and captured the reference, release the synchronous code.
+            await syncSemaphore.signal()
+            // Wait for the synchronous setup to complete and the reference to be nil'd out.
+            await asyncSemaphore.wait()
+            // Retain the unsafe counter until the task is completed.
+            _ = reference
+        }
+        // Wait for the asynchronous task to start.
+        await syncSemaphore.wait()
+        referenceHolder.reference = nil
+        XCTAssertNotNil(weakReference)
+        // Allow the enqueued task to complete.
+        await asyncSemaphore.signal()
+        // Make sure the task has completed.
+        await systemUnderTest.await { /* Drain the queue */ }
+        XCTAssertNil(weakReference)
+    }
+
+    func test_await_sendsEventsInOrder() async {
+        let counter = Counter()
+        for iteration in 1...1_000 {
+            systemUnderTest.async {
+                await counter.incrementAndExpectCount(equals: iteration)
+            }
+
+            guard iteration % 25 == 0 else {
+                // Keep sending async events to the queue.
+                continue
+            }
+
+            await systemUnderTest.await {
+                let count = await counter.count
+                XCTAssertEqual(count, iteration)
+            }
+        }
+        await systemUnderTest.await { /* Drain the queue */ }
+    }
+
+    func test_await_canReturn() async {
+        let expectedValue = UUID()
+        let returnedValue = await systemUnderTest.await { expectedValue }
+        XCTAssertEqual(expectedValue, returnedValue)
+    }
+
+    func test_await_canThrow() async {
+        struct TestError: Error, Equatable {
+            private let identifier = UUID()
+        }
+        let expectedError = TestError()
+        do {
+            try await systemUnderTest.await { throw expectedError }
+        } catch {
+            XCTAssertEqual(error as? TestError, expectedError)
+        }
+    }
+
+    // MARK: Private
+
+    private var systemUnderTest = AsyncQueue()
+
+    // MARK: - Counter
+
+    private actor Counter {
+        func incrementAndExpectCount(equals expectedCount: Int) {
+            increment()
+            XCTAssertEqual(expectedCount, count)
```

Comment:
Nit: would be nice to pass through the file and line number so any XCTest failure is easier to debug.

Reply:
b34e05a